### PR TITLE
clarify use of when

### PR DESCRIPTION
**Pull request by rosieyohannan**
fixes issue #4212

### CODE REVIEW

File: jekyll/_cci2/configuration-reference.md [hunk 1]
```diff
@@ -1533,7 +1533,14 @@ workflows:
 
 ##### **Using `when` in Workflows**
 
-With CircleCI configuration v2.1, you may use a `when` clause (the inverse clause `unless` is also supported) under a workflow declaration with a truthy or falsy value to determine whether or not to run that workflow. The most common use of `when` is with CircleCI API v2 pipeline triggering with parameters.
+With CircleCI v2.1 configuration, you may use a `when` clause (the inverse clause `unless` is also supported) under a workflow declaration with a truthy or falsy value to determine whether or not to run that workflow. The most common use of `when` is with [CircleCI API v2 pipeline triggering](https://circleci.com/docs/api/v2/#trigger-a-new-pipeline) with parameters. Truthy/falsy calues can be booleans, numbers, and strings. Falsy would be any of: false, 0, empty string, nil, and NaN. Everything else would be truthy.
+
+Key | Required | Type | Description
```
Comment (rosieyohannan) [inline jekyll/_cci2/configuration-reference.md:1538]: I need some help filling in this table 💥 😬

File: jekyll/_cci2/pipeline-variables.md [hunk 2]
```diff
@@ -166,11 +166,11 @@ Pipeline parameters which are defined in configuration are always in scope, with
 
 ## Conditional Workflows
 
-New as of June 2019, you can use a `when` clause (we also support the inverse clause `unless`) under a workflow declaration with a boolean value to decide whether or not to run that workflow.
+Use the `when` clause (or the inverse clause `unless`) under a workflow declaration, with a truthy or falsy value, to decide whether or not to run that workflow. Truthy/falsy calues can be booleans, numbers, and strings. Falsy would be any of: false, 0, empty string, nil, and NaN. Everything else would be truthy.
```
Comment (rosieyohannan) [inline jekyll/_cci2/pipeline-variables.md:169]:
```suggestion
Use the `when` clause (or the inverse clause `unless`) under a workflow declaration, with a truthy or falsy value, to decide whether or not to run that workflow. Truthy/falsy values can be booleans, numbers, and strings. Falsy would be any of: false, 0, empty string, nil, and NaN. Everything else would be truthy.
```

File: jekyll/_cci2/pipeline-variables.md [hunk 3]
```diff
@@ -166,11 +166,11 @@ Pipeline parameters which are defined in configuration are always in scope, with
 
 ## Conditional Workflows
 
-New as of June 2019, you can use a `when` clause (we also support the inverse clause `unless`) under a workflow declaration with a boolean value to decide whether or not to run that workflow.
+Use the `when` clause (or the inverse clause `unless`) under a workflow declaration, with a truthy or falsy value, to decide whether or not to run that workflow. Truthy/falsy values can be booleans, numbers, and strings. Falsy would be any of: false, 0, empty string, nil, and NaN. Everything else would be truthy.
 
 The most common use of this construct is to use a pipeline parameter as the value, allowing an API trigger to pass that parameter to determine which workflows to run.
 
-Below is an example configuration using two different pipeline parameters, one used to drive whether a particular workflow will run and another to determine if a particular step will run.
+Below is an example configuration using a pipeline parameter `run_integration_tests` to drive whether the workflow `integration_tests` will run.
```
Comment (rosieyohannan) [inline jekyll/_cci2/pipeline-variables.md:173]:
```suggestion
Below is an example configuration using the pipeline parameter `run_integration_tests` to drive whether the workflow `integration_tests` will run.
```

File: jekyll/_cci2/configuration-reference.md (hunk 1 above)
Comment (sulami) [reply jekyll/_cci2/configuration-reference.md:1538]: I'm not quite sure where these rows come from?

**Pull request merged**